bls12-381/src/main/java/org/hyperledger/besu/nativelib/bls12_381/LibEthPairings.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package org.hyperledger.besu.nativelib.bls12_381;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+import com.sun.jna.ptr.IntByReference;
+
+public class LibEthPairings implements Library {
+  public static final boolean ENABLED;
+
+  static {
+    boolean enabled;
+    try {
+      Native.register(LibEthPairings.class, "eth_pairings");
+      enabled = true;
+    } catch (final Throwable t) {
+      enabled = false;
+    }
+    ENABLED = enabled;
```

Comment:
why do we need this intermediate boolean `enabled`? something weird about `static` blocks?

Reply:
the field is final, and the compiler analysis considers catch blocks as a possible second setting because it doesn't calculate where in the try block the exception may be thrown.